packages/types/src/type/ExtrinsicEra.ts

```diff
@@ -2,33 +2,152 @@
 // This software may be modified and distributed under the terms
 // of the Apache-2.0 license. See the LICENSE file for details.
 
-import { AnyU8a } from '../types';
+import EnumType from '../codec/EnumType';
+import {isHex, isU8a, hexToU8a, isObject} from '@polkadot/util';
+import Tuple from '../codec/Tuple';
+import U64 from '../primitive/U64';
+import U8 from '../primitive/U8';
+import U32 from '../primitive/U32';
+import U16 from '../primitive/U16';
+import { IExtrinsicEra, AnyU8a } from '../types';
+import BlockNumber from "@polkadot/types/type/BlockNumber";
+import U8a from "@polkadot/types/codec/U8a";
 
-import { u8aToU8a } from '@polkadot/util';
+interface EraMethod {
+  startBlockNumber: BlockNumber;
+  endBlockNumber: BlockNumber;
+}
+
+export default class ExtrinsicEra extends EnumType<ImmortalEra | MortalEra> implements IExtrinsicEra {
+  constructor (value?: any, index?: number) {
+    super({ ImmortalEra, MortalEra }, value, index);
+  }
+
+  /**
+   * @description Returns the item as a [[ImmortalEra]]
+   */
+  get immortalEra (): ImmortalEra {
+    return this.value as ImmortalEra;
+  }
+
+  /**
+   * @description Returns the item as a [[MortalEra]]
+   */
+  get mortalEra (): MortalEra {
+    return this.value as MortalEra;
+  }
+
+  /**
+   * @description Encodes the value as a Uint8Array as per the parity-codec specifications
+   * @param isBare true when the value has none of the type-specific prefixes (internal)
+   */
+  toU8a (isBare?: boolean): Uint8Array {
+    if (this.index === 0 ) {
+      return super.toU8a()
+    } else {
+      return this.mortalEra.toU8a(isBare);
+    }
+  }
+}
+
+export class ImmortalEra extends U8a {
+  constructor(value?: AnyU8a) {
+    super(value);
+  }
+}
 
-import U8a from '../codec/U8a';
+export type MortalEraValue = [U8, U8];
 
 /**
- * @name ExtrinsicEra
+ * @name MortalEra
  * @description
- * The era for an extrinsic, indicating either a mortal or immortal extrinsic
+ * The MortalEra for an extrinsic, indicating period and phase
  */
-export default class ExtrinsicEra extends U8a {
-  constructor (value?: AnyU8a) {
-    super(
-      ExtrinsicEra.decodeExtrinsicEra(value)
-    );
+export class MortalEra extends Tuple {
+  constructor (value?: any) {
+    super({
+      period : U8, phase : U8
+    }, MortalEra.decodeMortalEra(value));
+  }
+
+  private static decodeMortalEra (value: EraMethod | Uint8Array | string): MortalEraValue {
+    if (isHex(value)) {
+      return MortalEra.decodeMortalEra(hexToU8a(value.toString()));
+    } else if (isU8a(value)) {
+      const u8a = value;
+      const first = u8a.subarray(0, 1);
+      let second = u8a.subarray(1, 2);
+      const encoded = new U64(new U64(first).toNumber() + (new U64(second).toNumber() << 8));
```

Comment:
you can just use them as type number. (no point wrap them with U64 but toNumber() every time use them)